### DISCUSSION TITLE
Help with large attachments

### DISCUSSION THREAD

**Discussion by WISDTech** [jsuto/piler#186] [Q&A] [answered]
I installed piler 1.4.6 on Ubuntu 24.04.  Took a while but got it complete.  All our 365 email are showing up as expected.  According to Piler we processed 27,000 emails in 24 hours..  Everything appears to be working until we noticed emails with large attachments were being rejected.  I edited piler.conf and upped the number for max_message_size= but I'm not sure what this variable means or if it worked.  Unlimited attachment size would be preferred.  The biggest attachment right now is around 40 mb.  When I try to view the emails in auditor I get the popup " piler.mydomain.com:8080 says Problem retrieving XML data:error"  I have been stumped for two days now.  I noticed that any email with an attachment around the size of 20 mb or less have no issues.  I need help figuring out how to resolve this large attachment issue, and it would be nice if there was documentation on all the variables in the piler.conf file.  Thanks in advance for any help.

Installed on a VM with 
4 TB drive
16 G memory 11% used
several CPU 2% utilized

Any additional tips for a large volume setup is appreciated.

**Comment by WISDTech** [comment]
I found the following regarding max message size.  I'll test this tomorrow.

max_message_size sets the max. message size that piler-smtp accepts The default value is 50000000 bytes (~50 MB)

max_smtp_memory limits the max. available memory to buffer messages. when this limit is exceeded, no new emails will be accepted until the used memory for all in progress emails decreases below this level.
  The default value is 500000000 bytes (~500 MB)


Still getting the error when selecting an email with over 20 mb attachment.

**Comment by jsuto** [comment]
> emails with large attachments were being rejected.

Elaborate what "rejected" means. It seems that you refer to the piler-smtp daemon where the memory limits you mentioned apply, however you experience issues when clicking on large messages on the gui.

**Reply by WISDTech** [reply]
Thank you for replying so quickly.  This implementation is for a school district in which our very old version of MailPiler installed by someone else has reached its limit.

The exact error returned by Office 365 admin portal is "550 5.0.350 Remote server returned an error -> 552 Too big email".  I will test today if the changes I made in the earlier post fixes this issue.

In addition to the message size limit being 50 mb by default, I currently am not able to view any email with an attachment larger that 20 mb.  A popup message appears "piler.domain.com:8080 says Problem retrieving XML data:error".  Of course I changed the domain name.  I have to click the OK button to move on.  If I check the check box next to the email then try to download the email I get the error page "This page isn't working piler.domain.com is currently unable to handle the request.  HTTP ERROR 500".  I hit the back button to return to the auditing page.  I am able to download any other email.

I have installed two Piler servers one being for staff the other for student email retention.  I keep the exact same configuration for both.  Both are doing the same thing.

**Reply by jsuto** [reply, marked as answer]
Figure out the max message size your MX servers accept, and adjust piler's smtp memory limits accordingly.

Regarding the problem in the GUI, you may need to adjust the php memory limits as well. Check **memory_limit**, its value must be bigger than your max message size. Try setting double the size.

**Reply by WISDTech** [reply]
That worked!  Thank you.  I don't know why I didn't thing of this.  I had to up the memory limit for PHP several times in the past for other projects.  I have added this to my installation instructions.  I still haven't tested sending a large attachment yet but I bet it works.

**Reply by WISDTech** [reply]
Yes.  I am now receiving larger than 50 mb messages.  Thanks again for your help.

**Reply by 0xD0M1M0** [reply]
> Figure out the max message size your MX servers accept, and adjust piler's smtp memory limits accordingly.
> 
> Regarding the problem in the GUI, you may need to adjust the php memory limits as well. Check **memory_limit**, its value must be bigger than your max message size. Try setting double the size.

I have a 25MB e-mail and I am getting as well the error in the GUI "Problem retrieving XML data:error". I am using the docker installation method. Where can I edit the php memory limit?

**Comment by jsuto** [comment]
In /etc/php/8.3/fpm/php.ini locate the memory_limit value, then run ```service php8.3-fpm restart```

**Reply by 0xD0M1M0** [reply]
Perfect. Doubling the value worked. Köszönöm!
